smartconfig/file.py

```diff
@@ -0,0 +1,55 @@
+import yaml
+
+from smartconfig import _registry
+from smartconfig.constructors import _ref_constructor
+from smartconfig.typehints import YAMLStructure, _FilePath
+
+
+def _recursively_update_mapping(source: YAMLStructure, dest: YAMLStructure) -> YAMLStructure:
+    """
+    Recursively update the dest mapping with source.
+
+    If a key contains a dot, it will be considered as two nested dictionaries.
+
+    Args:
+        source: The update to apply.
+        dest: The mapping to update.
+
+    Returns:
+        The modified dest mapping.
```

Comment:
Shouldn't this be:
```py
Returns:
    YAMLStructure: The modified dest mapping.
```

Reply:
That shouldn’t be necessary since we have a return typehint. Thanks though!